client/keystore/keystore.go

```diff
@@ -18,119 +24,269 @@ const (
 	NewPassphrasePromptRepeat = "Repeat passphrase:"
 
 	PassphrasePrompt = "Enter passphrase:"
+
+	accountDir = "accounts.ldb"
+	keysDir    = "keys"
+
+	DirFlag = "directory"
 )
 
 var ks *keystore.KeyStore
 
 // initialize a keystore in the specified directory
-func InitKeystore(dir string) {
+func InitKeystore() {
+	dir := getDir(keysDir)
```

Comment:
Rather than open the account db in every function, we can open it here along with the keystore and save a reference or abort if the database is corrupted

Reply:
Ah, the tricky thing though would be calling db.close() when done.